### ISSUE TITLE
Option to force delete resources

### ISSUE THREAD

**Issue by kvaps** [LINBIT/linstor-client#17]
It is really needed,

I still have some resources which can't be deleted at all. See https://github.com/LINBIT/linstor-server/issues/32#issuecomment-496170497 for example.

Sometimes the only way to delete broken resource is to `lost node` then redeploy it from beginning and all the good resources on it.

This is madness. We all understand that `--force` is bad, but sometimes it is really required.
Add `--yes-i-really-mean-it` option or something else for make people thinking, but please return `--force` option back.

**Comment by raltnoeder** [comment]
No, it's not. If the resource cannot be deleted automatically, but is marked for deletion, then the resource is supposed to disappear as soon as it has been cleaned up manually. If it doesn't, then it has either not been cleaned up properly, and the automatic procedure cannot clean it up, or there is a bug that needs to be fixed.
But then we'll fix the bug instead of turning the system into a gigantic mess again where administrators constantly shoot themselves in their own foot, because we know from experience that this is exactly what will happen as soon as we allow administrators to pretend that a resource that is still there has been deleted.
And we are the people who get the complaints when they lose their data again.

Additionally, it would be great if the environment that LINSTOR needs to run in were not the totally unreliable unstandardized chaotic mess that it unfortunately is, where every other version of the Linux kernel, the LVM utilties, the ZFS utilities and essentially every other system utility breaks in different ways every other day, and where there are no documented and unambiguous APIs to control what the system is supposed to do in an at least somewhat deterministic manner.
So my suggestion is to take this one level higher and complain to the people who maintain the stuff that keeps failing outside of the linstor satellite. Because that's where the actual madness is coming from.

**Comment by raltnoeder** [comment]
What will be in one of the following releases is a command that declares an entire storage pool lost, so that resources which are part of a storage pool that does not exist anymore can be deleted successfully.

**Comment by kvaps** [comment]
Hi @raltnoeder, thanks for the detailed response. I understand your point of view, but --force option it is still better than try to delete information about the resource from the database or delete whole node with all resources (including good ones), then create all of them again.

I agree that it might be problematic for your support. But there is always alternative options are exist, eg: allow force removing only via debug console of linstor-controller, or requiring `--yes-i-understand-that-i-lose-support-by-executing-this-action` option, anything.

I know that bug should be solved, but due to lack of information and unpopularity I don't believe that it can be solved soon. Right now it is little blocking development, and I just want  to get rid of the annoying resource.

> What will be in one of the following releases is a command that declares an entire storage pool lost, so that resources which are part of a storage pool that does not exist anymore can be deleted successfully.

Sounds good, but what to do with the good resources inside the same pool?

**Comment by raltnoeder** [comment]
If the storage pool is still there and working, then what prevents you from simply deleting the resource?
Resource deletion stops the DRBD resource, then deletes the backend storage volumes and the configuration files. Have you read the bug report and determined what prevents it from doing so?
Is the DRBD resource still configured?
Is the backend storage volume still there?
Is the <resourcename>.res file still there?
What is the exact error that was reported when the resource deletion failed?

**Comment by petrsmetana** [comment]
Hello, is there any other option to force remove the resources. Over time, we have accumulated a rather large bunch of resources that can not be deleted by the standard way, because they are just
stucked on deleting with no errors:

See issue - https://github.com/LINBIT/linstor-server/issues/90

```
root@linstor-controller-0:/# linstor r l | grep one | grep -v Diskless | grep -v UpToDate
| one-image-2055                                 | m13c37 | 8412 |        |           DELETING |
| one-vm-5561-disk-0                             | m10c37 | 8647 |        |           DELETING |
| one-vm-5760-disk-0                             | m10c35 | 8616 |        |           DELETING |
| one-vm-6182-disk-0                             | m10c9  | 8548 |        |           DELETING |
| one-vm-6242-disk-0                             | m10c2  | 8538 |        |           DELETING |
| one-vm-6571-disk-0                             | m13c41 | 7424 |        |           DELETING |
| one-vm-6928-disk-0                             | m10c2  | 8512 |        |           DELETING |
| one-vm-6928-disk-0                             | m10c8  | 8512 |        |           DELETING |
| one-vm-6948-disk-2                             | m10c8  | 8507 |        |           DELETING |
| one-vm-7056-disk-0                             | m10c9  | 8490 |        |           DELETING |
| one-vm-7059-disk-0                             | m10c8  | 8489 |        |           DELETING |
| one-vm-7247-disk-0                             | m13c41 | 7921 |        |           DELETING |
| one-vm-7432-disk-0                             | m13c41 | 7855 |        |           DELETING |
| one-vm-7569-disk-0                             | m13c41 | 7988 |        |           DELETING |
| one-vm-7596-disk-0                             | m13c41 | 7786 |        |           DELETING |
| one-vm-7748-disk-2                             | m13c41 | 7714 |        |           DELETING |
| one-vm-7792-disk-0                             | m13c41 | 8026 |        |           DELETING |
| one-vm-7797-disk-0                             | m13c41 | 7624 |        |           DELETING |
| one-vm-8001-disk-0                             | m13c41 | 7520 |        |           DELETING |
| one-vm-8173-disk-2                             | m10c2  | 8455 |        |           DELETING |
```


Neither restarting of linstor-controller and linstor-satellite does not solves problem.

* Is the DRBD resource still configured?
No, drbdadm status does not showing this resource

* Is the backend storage volume still there?
No, this problem also affects diskless resources

* Is the .res file still there?
No, resource remaing DELETING even after the reboot of whole node

**Comment by raltnoeder** [comment]
@ghernadi mentioned in another thread that restarting the controller/satellite apparently does not clean up resources that are still marked for deletion. Issuing the delete command again while the satellite is online normally does, or it generates an error report. The resources neither being deleted nor having an error report generated indicates that no attempt was made to delete those resources.
So to summarize: I'd recommend to issue another `resource delete` or `resource-definition delete` command while the satellite is online to trigger a retry of the deletion.
